tsdb/tsm1/engine_schema.go

```diff
@@ -368,6 +461,14 @@ func (e *Engine) tagKeysPredicate(ctx context.Context, orgBucket []byte, start,
 		}
 
 		for _, iter := range iters {
+			// Always check context before performing a tsm file scan
```

Comment:
Same here, given seek is binary search, checking every 64 keys of the outer loop should suffice.